framework/src/utils/SlepcSupport.C

```diff
@@ -951,6 +951,13 @@ mooseSlepcEigenFormFunctionB(SNES snes, Vec x, Vec r, void * ctx)
 
     ierr = MatMult(B, x, r);
     CHKERRQ(ierr);
+
+    if (eigen_problem->bxNormProvided())
+    {
+      // User has provided a postprocessor. We need it updated
+      updateCurrentLocalSolution(eigen_nl.sys(), x);
+      eigen_problem->execute(EXEC_LINEAR);
```

Comment:
I realized that we do not have a check to make sure `execute_on` of the bx_norm postprocessor includes linear. Care to add it in `EigenProblem`?